google-cloud-redis/src/test/java/com/google/cloud/redis/v1/it/ITSystemTest.java

```diff
@@ -103,7 +105,14 @@ public void testUpdateInstance() throws ExecutionException, InterruptedException
             .build();
     UpdateInstanceRequest updateInstanceRequest =
         UpdateInstanceRequest.newBuilder().setInstance(instance).setUpdateMask(updateMask).build();
-    Instance actualInstance = client.updateInstanceAsync(updateInstanceRequest).get();
-    assertEquals(memorySizeGb, actualInstance.getMemorySizeGb());
+    OperationFuture<Instance, OperationMetadata> future =
+        client.updateInstanceAsync(updateInstanceRequest);
+    while (true) {
+      if (future.isDone()) {
+        Instance updatedInstance = future.get();
+        assertEquals(memorySizeGb, updatedInstance.getMemorySizeGb());
+        break;
+      }
+    }
```

Comment:
I don't understand, why would you need a while loop. 

As OperationFuture#get is already blocking then what is it happening in the original test case to make it flaky?

Reply:
Also I don't think this is fixing the core issue. For whatever reason the tests were flaky it would still be flaky and might still fail at Line 113.

Reply:
how you decide it would be fail at Line 113?

Reply:
If the original test is flaky that means it would only fail at assertEquals(). In this case also we are not doing anything other than asserting memorySize once `future#isDone` is `true`.

Reply:
do you know what was the exact problem?